satellite/contact/contact_test.go

```diff
@@ -17,19 +22,27 @@ func TestSatelliteContactEndpoint(t *testing.T) {
 	testplanet.Run(t, testplanet.Config{
 		SatelliteCount: 1, StorageNodeCount: 1, UplinkCount: 0,
 	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
-		satDossier := planet.Satellites[0].Local()
 		nodeDossier := planet.StorageNodes[0].Local()
+		ident := planet.StorageNodes[0].Identity
 
-		conn, err := planet.StorageNodes[0].Transport.DialNode(ctx, &satDossier.Node)
-		require.NoError(t, err)
-		defer ctx.Check(conn.Close)
-
-		resp, err := pb.NewNodeClient(conn).Checkin(ctx, &pb.CheckinRequest{
-			Address:  nodeDossier.GetAddress(),
+		grpcPeer := peer.Peer{
+			Addr: &net.TCPAddr{
+				IP:   net.ParseIP(nodeDossier.Address.GetAddress()),
+				Port: 5,
+			},
+			AuthInfo: credentials.TLSInfo{
+				State: tls.ConnectionState{
+					PeerCertificates: []*x509.Certificate{ident.Leaf, ident.CA},
+				},
+			},
+		}
+		peerCtx := peer.NewContext(ctx, &grpcPeer)
+		resp, err := planet.Satellites[0].Contact.Endpoint.Checkin(peerCtx, &pb.CheckinRequest{
```

Comment:
Would be awesome to see that the node info in the overlay is updated when this endpoint is hit

Reply:
for sure, good idea. yeah im going to add more tests next as part of ticket 2571.